examples/silentpayments.c

```diff
@@ -0,0 +1,503 @@
+    0x02, 0xee, 0x97, 0xdf, 0x83, 0xb2, 0x54, 0x6a,
+    0xf5, 0xa7, 0xd0, 0x62, 0x15, 0xd9, 0x8b, 0xcb,
+    0x63, 0x7f, 0xe0, 0x5d, 0xd0, 0xfa, 0x37, 0x3b,
+    0xd8, 0x20, 0xe6, 0x64, 0xd3, 0x72, 0xde, 0x9a, 0x01
+};
+static unsigned char bob_address[2][33] = {
+    {
+        0x02, 0x15, 0x40, 0xae, 0xa8, 0x97, 0x54, 0x7a,
+        0xd4, 0x39, 0xb4, 0xe0, 0xf6, 0x09, 0xe5, 0xf0,
+        0xfa, 0x63, 0xde, 0x89, 0xab, 0x11, 0xed, 0xe3,
+        0x1e, 0x8c, 0xde, 0x4b, 0xe2, 0x19, 0x42, 0x5f, 0x23
+    },
+    {
+        0x02, 0x3e, 0xff, 0xf8, 0x18, 0x51, 0x65, 0xea,
+        0x63, 0xa9, 0x92, 0xb3, 0x9f, 0x31, 0xd8, 0xfd,
+        0x8e, 0x0e, 0x64, 0xae, 0xf9, 0xd3, 0x88, 0x07,
+        0x34, 0x97, 0x37, 0x14, 0xa5, 0x3d, 0x83, 0x11, 0x8d
+    }
+};
+static unsigned char carol_scan_key[32] = {
+    0x04, 0xb2, 0xa4, 0x11, 0x63, 0x5c, 0x09, 0x77,
+    0x59, 0xaa, 0xcd, 0x0f, 0x00, 0x5a, 0x4c, 0x82,
+    0xc8, 0xc9, 0x28, 0x62, 0xc6, 0xfc, 0x28, 0x4b,
+    0x80, 0xb8, 0xef, 0xeb, 0xc2, 0x0c, 0x3d, 0x17
+};
+static unsigned char carol_address[2][33] = {
+    {
+        0x03, 0xbb, 0xc6, 0x3f, 0x12, 0x74, 0x5d, 0x3b,
+        0x9e, 0x9d, 0x24, 0xc6, 0xcd, 0x7a, 0x1e, 0xfe,
+        0xba, 0xd0, 0xa7, 0xf4, 0x69, 0x23, 0x2f, 0xbe,
+        0xcf, 0x31, 0xfb, 0xa7, 0xb4, 0xf7, 0xdd, 0xed, 0xa8
+    },
+    {
+        0x03, 0x81, 0xeb, 0x9a, 0x9a, 0x9e, 0xc7, 0x39,
+        0xd5, 0x27, 0xc1, 0x63, 0x1b, 0x31, 0xb4, 0x21,
+        0x56, 0x6f, 0x5c, 0x2a, 0x47, 0xb4, 0xab, 0x5b,
+        0x1f, 0x6a, 0x68, 0x6d, 0xfb, 0x68, 0xea, 0xb7, 0x16
+    }
+};
+
+/** Labels
+ *
+ *  The structs and call back function are implemented here as a demonstration
+ *  of how the label lookup callback is meant to query a label cache and return
+ *  the label tweak when a match is found. This is for demonstration purposes
+ *  only and not optimized. In a production usecase, it is expected that the
+ *  caller will be using a much more performant method for storing and querying
+ *  labels.
+ *
+ *  Recipients not using labels can ignore these steps and simply pass `NULL`
+ *  for the label_lookup and label_context arguments:
+ *
+ *      secp256k1_silentpayments_recipient_scan_outputs(..., NULL, NULL);
+ */
+
+struct label_cache_entry {
+    unsigned char label[33];
+    unsigned char label_tweak[32];
+};
+
+struct labels_cache {
+    size_t entries_used;
+    struct label_cache_entry entries[5];
+};
+
+const unsigned char* label_lookup(
+    const unsigned char* label33,
+    const void* cache_ptr
+) {
+    const struct labels_cache* cache = (const struct labels_cache*)cache_ptr;
+    size_t i;
+    for (i = 0; i < cache->entries_used; i++) {
+        if (secp256k1_memcmp_var(cache->entries[i].label, label33, 33) == 0) {
+            return cache->entries[i].label_tweak;
+        }
+    }
+    return NULL;
+}
+
+int main(void) {
+    enum { N_INPUTS = 2, N_OUTPUTS = 3 };
+    unsigned char randomize[32];
+    unsigned char xonly_print[32];
```

Comment:
https://github.com/bitcoin-core/secp256k1/pull/1519/commits/5c546e2874012dccc40a3e91ed2502c538a00356: nit: Can we change `xonly_print` to `serialized_xonly_pk`? or maybe add a comment `/* Serialized xonly pk */`

Reply:
I agree that `_print` is a bit redundant, but shouldn't `unsigned char ... [32]` make it clear that this is the serialization, no?

Reply:
To you and me, yes, but to the uninitiated, spelling it out in the variable name might help. This code is meant to be read by anyone trying to use the silent-payments API, so it helps to make things clearer for beginners. That said, I don't have any strong feelings about this; it's just a "nit."